src/jvm/test/org/jetbrains/kotlinx/lincheck_test/representation/ObstructionFreedomViolationEventsCutTest.kt

```diff
@@ -0,0 +1,81 @@
+/*
+ * Lincheck
+ *
+ * Copyright (C) 2019 - 2025 JetBrains s.r.o.
+ *
+ * This Source Code Form is subject to the terms of the
+ * Mozilla Public License, v. 2.0. If a copy of the MPL was not distributed
+ * with this file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+package org.jetbrains.kotlinx.lincheck_test.representation
+
+import org.jetbrains.kotlinx.lincheck.annotations.Operation
+import org.jetbrains.kotlinx.lincheck.checkImpl
+import org.jetbrains.kotlinx.lincheck.isInTraceDebuggerMode
+import org.jetbrains.kotlinx.lincheck.strategy.managed.modelchecking.ModelCheckingOptions
+import org.jetbrains.kotlinx.lincheck_test.util.checkLincheckOutput
+import org.junit.Assume.assumeFalse
+import org.junit.Before
+import org.junit.Test
+import java.util.concurrent.atomic.AtomicReference
+
+/**
+ * Checks that spin-cycle repeated events are cut in case of obstruction freedom violation
+ */
+class ObstructionFreedomViolationEventsCutTest {
```

Comment:
why this is required to be put outside of `SpinlockEventsCutTests.kt` file? It logically belongs to other `events cut` tests